src/Extension/Mention/MentionExtension.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the league/commonmark package.
+ *
+ * (c) Colin O'Dell <colinodell@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace League\CommonMark\Extension\Mention;
+
+use League\CommonMark\ConfigurableEnvironmentInterface;
+use League\CommonMark\Extension\ExtensionInterface;
+
+final class MentionExtension implements ExtensionInterface
+{
+    public function register(ConfigurableEnvironmentInterface $environment)
+    {
+        $registeredMentions = $environment->getConfig('registered_mentions', []);
+
+        // Retrieve mentions, excluding already registered mentions.
```

Comment:
We can probably avoid this de-duplication logic since this `register` method is only ever called once per extension instance.  Yes, somebody could potentially add multiple `MentionExtension`s to their environment, but the only negative result would be a slight performance decrease which I'm okay with.

Reply:
Ok.